src/components/PurchaseOrder/POForm.js

```diff
@@ -178,11 +179,25 @@ class POForm extends Component {
   }
 
   onChangeTemplate = (e, value) => {
-    const { change, dispatch, parentResources, stripes } = this.props;
+    const { change, dispatch, reset, parentResources, stripes } = this.props;
     const templateValue = getOrderTemplateValue(parentResources, value);
     let form = get(stripes.store.getState(), 'form', {});
+    const { FormPO: { values: { poNumber } } } = form;
 
+    dispatch(reset());
```

Comment:
do you need to dispatch reset when you dispatch changes mostly for all fields?

Reply:
It looks like `reset()` works for form options (dirty, etc.), but fields has to be reset manually

Reply:
@NikitaSedyx, good note, thanks. Here was a problem with it, `dispatch(reset())` work only for creating form, but not for editing mode, so I add `dispatch(change())` for main fields to reset them
I rechecked it, deleted this string and everything works as with reset, so, yes, it could be missed